### PR TITLE
Add `--fix` support to `order-in-*` rules

### LINKED ISSUE #161: Add `fix` for order-in-* rules
I'd like to propose adding fixer for `order-in-*` rules. AFAIK the common logic for those rules can be found in `property-order` util, and there's the only `context.report()` call. The good thing is that the hard work is done already - finding the token before which the current one should be inserted.
I'd like to handle this task.

What do you think?

### PR DESCRIPTION
**Pull request by PrzemoRevolve**
Resolves #161

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ node_modules
 coverage
 npm-debug.log
 *.swp
+.vscode
```
Comment (Turbo87) [inline .gitignore]: this change is unrelated to the PR and stuff like that should be added to your global gitignore file instead

File: README.md [hunk 2]
```diff
@@ -88,6 +88,13 @@ The `--fix` option on the command line automatically fixes problems reported by
 | :white_check_mark: | [no-on-calls-in-components](./docs/rules/no-on-calls-in-components.md) | Prevents usage of `on` calls in components |
 
 
+### Ember Object
+
+|    | Rule ID | Description |
+|:---|:--------|:------------|
+| :white_check_mark: | [avoid-leaking-state-in-ember-objects](./docs/rules/avoid-leaking-state-in-ember-objects.md) | Avoids state leakage |
+
+
```
Comment (Turbo87) [inline README.md]: unrelated too

File: lib/recommended-rules.js [hunk 3]
```diff
@@ -7,6 +7,7 @@
 module.exports = {
   "ember/alias-model-in-controller": "error",
   "ember/avoid-leaking-state-in-components": "error",
+  "ember/avoid-leaking-state-in-ember-objects": "error",
```
Comment (Turbo87) [inline lib/recommended-rules.js]: unrelated too

File: tests/lib/rules/order-in-controllers.js [hunk 4]
```diff
@@ -223,5 +223,37 @@ eslintTester.run('order-in-controllers', rule, {
...
       }],
     },
+    {
+      code: `export default Controller.extend({
+        queryParams: [],
+        currentUser: service(),
+      });`,
+      output: `export default Controller.extend({
+        currentUser: service(),
+      queryParams: [],
```
Comment (Turbo87) [inline tests/lib/rules/order-in-controllers.js]: looks like indentation is broken here?

File: tests/lib/rules/order-in-routes.js [hunk 5]
```diff
@@ -361,6 +361,139 @@ eslintTester.run('order-in-routes', rule, {
...
+        customProp: "test",
+        /**
+         * beforeModel block comment
+         */
+        beforeModel() {},
+        model() {},
+        /**
+         * actions block comment
+         */
+        actions: {},
```
Comment (Turbo87) [inline tests/lib/rules/order-in-routes.js]: can you add a test where actions with comment is above e.g. beforeModel with a comment to verify that comments are moved correctly too?

File: .gitignore (hunk 1 above)
Comment (PrzemoRevolve) [reply .gitignore]: Oh, sure

File: README.md (hunk 2 above)
Comment (PrzemoRevolve) [reply README.md]: Yeah... must have been added automatically when I ran `yarn run update`

File: tests/lib/rules/order-in-controllers.js (hunk 4 above)
Comment (PrzemoRevolve) [reply tests/lib/rules/order-in-controllers.js]: Yes, I did not fix that, because the `[indent]` rule can do that. This is indented wrong because `currentUser` used to be the last property, so this is the whitespace that was copied. I didn't want to add another `if` statement to fix that, because people can have different indentation settings.

File: tests/lib/rules/order-in-routes.js (hunk 5 above)
Comment (PrzemoRevolve) [reply tests/lib/rules/order-in-routes.js]: Of course

File: tests/lib/rules/order-in-routes.js [hunk 6]
```diff
@@ -361,6 +361,168 @@ eslintTester.run('order-in-routes', rule, {
...
+         */
+        beforeModel() {}
+      });`,
+      output: `export default Route.extend({
+        customProp: "test",
+        /**
+         * beforeModel block comment
+         */
+        beforeModel() {},
+      /**
```
Comment (PrzemoRevolve) [inline tests/lib/rules/order-in-routes.js]: Ok, so the indentation is wrong here as well, but `[indent]` rule doesn't fix this, so I have to think how to prevent that from happening.

File: tests/lib/rules/order-in-routes.js (hunk 5 above)
Comment (PrzemoRevolve) [reply tests/lib/rules/order-in-routes.js]: I've added it below

**Comment by myasonik** [comment]
Is there anything left to do on this PR for it to get merged?

**Comment by PrzemoRevolve** [comment]
@myasonik I wrote this for v3, now I see that v5 is the latest. I could change my code to use the new API. What do you think?

**Comment by myasonik** [comment]
Merging this would really ease adoption of the rule in large codebases. It'd be a great addition to get it updated and merged.

**Comment by PrzemoRevolve** [comment]
The conflicts are resolved. Ready to be merged, if you'd ask me.

**Comment by Turbo87** [comment]
sorry for the long delay, and thanks a lot for working on this ❤️

**Comment by Turbo87** [comment]
released in v5.1.0 🎉
